### PR TITLE
feat: Add profilesClient, EndpointClient, and pipClient

### PR DESCRIPTION
**Pull request by jwtty**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add atmProfilesClient and atmEndpointsClient for hub-net-controller and publicIPAddressClient for member-net-controller.
The clients are used to the azure traffic manager integration feature.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [x] run `make reviewable` for basic local test
- [ ] Read and followed fleet-networking's [Code of conduct](https://github.com/Azure/fleet-networking/blob/main/CODE_OF_CONDUCT.md).
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

File: cmd/hub-net-controller-manager/main.go [hunk 1]
```diff
@@ -185,16 +190,22 @@ func main() {
...
-		// }
-		// cloudConfig.SetUserAgent("fleet-hub-net-controller-manager")
-		// klog.V(1).InfoS("Cloud config loaded", "config", cloudConfig)
+		klog.V(1).InfoS("Traffic manager feature is enabled, loading cloud config and creating azure clients", "cloudConfigFile", *cloudConfigFile)
+		cloudConfig, err := azure.NewCloudConfigFromFile(*cloudConfigFile)
+		if err != nil {
+			klog.ErrorS(err, "Unable to load cloud config", "file name", *cloudConfigFile)
+			exitWithErrorFunc()
+		}
+		cloudConfig.SetUserAgent("fleet-hub-net-controller-manager")
```
Comment (zhiying-lin) [inline cmd/hub-net-controller-manager/main.go:200]: better to add some logs here.

File: cmd/member-net-controller-manager/main.go [hunk 2]
```diff
@@ -314,15 +319,21 @@ func setupControllersWithManager(ctx context.Context, hubMgr, memberMgr manager.
...
-		// 	exitWithErrorFunc()
-		// }
-		// cloudConfig.SetUserAgent("fleet-member-net-controller-manager")
+		klog.V(1).InfoS("Traffic manager feature is enabled, loading cloud config and creating azure clients", "cloudConfigFile", *cloudConfigFile)
+		cloudConfig, err := azure.NewCloudConfigFromFile(*cloudConfigFile)
+		if err != nil {
+			klog.ErrorS(err, "Unable to load cloud config", "file name", *cloudConfigFile)
+			return err
+		}
+		cloudConfig.SetUserAgent("fleet-member-net-controller-manager")
```
Comment (zhiying-lin) [inline cmd/member-net-controller-manager/main.go:330]: ditto

File: cmd/member-net-controller-manager/main.go [hunk 3]
```diff
@@ -333,6 +344,7 @@ func setupControllersWithManager(ctx context.Context, hubMgr, memberMgr manager.
 		HubNamespace:                mcHubNamespace,
 		Recorder:                    memberMgr.GetEventRecorderFor(serviceexport.ControllerName),
 		EnableTrafficManagerFeature: *enableTrafficManagerFeature,
```
Comment (zhiying-lin) [inline cmd/member-net-controller-manager/main.go:346]: need to populate the resourceGroup too

File: cmd/hub-net-controller-manager/main.go [hunk 4]
```diff
@@ -203,3 +214,36 @@ func main() {
...
+	}
+	options, err := azclient.GetDefaultResourceClientOption(&cloudConfig.ARMClientConfig, factoryConfig)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get default resource client option: %w", err)
+	}
+
+	rateLimitPolicy := ratelimit.NewRateLimitPolicy(cloudConfig.Config)
+	if rateLimitPolicy != nil {
+		options.ClientOptions.PerCallPolicies = append(options.ClientOptions.PerCallPolicies, rateLimitPolicy)
+	}
```
Comment (zhiying-lin) [inline cmd/hub-net-controller-manager/main.go:237]:
```suggestion
	
	if rateLimitPolicy := ratelimit.NewRateLimitPolicy(cloudConfig.Config); rateLimitPolicy != nil {
		options.ClientOptions.PerCallPolicies = append(options.ClientOptions.PerCallPolicies, rateLimitPolicy)
	}
```

**Review by zhiying-lin — APPROVED** [review]

File: cmd/hub-net-controller-manager/main.go (hunk 4 above)
Comment (jwtty) [reply cmd/hub-net-controller-manager/main.go]: fixed.

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**